### PR TITLE
New tool 'advanced-py-prof'

### PR DESCRIPTION
**Pull request by jgfouca**
Allows for creation of callgraph dotgraph profiles.

Test suite: None
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: New tool

Code review: @jedwards4b

### CODE REVIEW

File: scripts/Tools/advanced-py-prof [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#! /bin/bash -e
+
+# Usage: advanced-py-prof <python exec and arguments>
+# Requires graphviz and gprof2dot
+
+DATE_STAMP=$(date "+%Y-%m-%d_%H%M%S")
+FILE_BASENAME=$(basename $1).${DATE_STAMP}
+python -m cProfile -o ${FILE_BASENAME}.prof "$@"
```
Comment (jedwards4b) [inline scripts/Tools/advanced-py-prof]: Can you add code to make sure that these externals are in the path and exit gracefully if they do not?

Comment (jgfouca) [inline scripts/Tools/advanced-py-prof]: Sure

Comment (jgfouca) [inline scripts/Tools/advanced-py-prof]: Done.

**Comment by jgfouca** [comment]
See if you approve the changes, then this should be ready.

**Pull request merged**